compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -480,8 +480,55 @@ impl<'ll, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'_, 'll, 'tcx> {
             }
 
             _ if name.as_str().starts_with("simd_") => {
+                // Unpack non-power-of-2 #[repr(packed)]
+                let mut loaded_args = Vec::new();
+                for (ty, arg) in arg_tys.iter().zip(args) {
+                    loaded_args.push(
+                        if ty.is_simd()
+                            && let OperandValue::Ref(place) = arg.val
+                        {
+                            let (size, elem_ty) = ty.simd_size_and_type(self.tcx());
+                            let elem_ll_ty = match elem_ty.kind() {
+                                ty::Float(f) => self.type_float_from_ty(*f),
+                                ty::Int(i) => self.type_int_from_ty(*i),
+                                ty::Uint(u) => self.type_uint_from_ty(*u),
+                                ty::RawPtr(_, _) => self.type_ptr(),
+                                _ => unreachable!(),
+                            };
```

Comment:
Is there a function for this, `Ty -> Type`?

Reply:
It would have to be defined in `cg_llvm` instead of on `Ty` or `TyKind`, since each codegen backend would have to accept them and convert it into its local "type".

Reply:
Yeah, I was wondering if this exists somewhere in the builder traits so I could be defined per backend.

Reply:
Hmm. I looked around and I think it should only live in this file, because that's a Very Dicey thing to do in the general case, but perfectly reasonable here, and all of the uses of the `from_ty` family are here anyways.

Reply:
( It's possible I'm wrong about the diceyness but I'm still ambiently unsure about whether this is the correct level of abstraction. )